File: AOJ/ALDS1_1_B.scala
object Main{
  def main(args: Array[String]) = {
    val sc = new java.util.Scanner(System.in)
    val a, b = sc.nextInt
    val x = math.max(a, b)
    val y = math.min(a, b)
    println(gcd(x,y))
  }
  def gcd(x: Int, y: Int) : Int = {
    if (y == 0) x else gcd(y, x%y)
  }
}

File: AOJ/ALDS/ALDS1_1_B.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
 
public class Main {
    static void gcd(int a, int b){
            int x = Math.max(a,b);
            int y = Math.min(a,b);
            int t = 0;
            while (y>0){
                t = x;
                x = y;
                y = t%y;
            }
            System.out.println(x);
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] line = br.readLine().split(" ");
        int a = Integer.parseInt(line[0]);
        int b = Integer.parseInt(line[1]);
        gcd(a,b);
    }
}
